### DISCUSSION TITLE
Codeium Plugin

### DISCUSSION THREAD

**Discussion by rrangelo** [apache/netbeans#5787] [Ideas]
Hello! I am a Netbeans user for 15 years, I would like to be able to use [Codeium AI](https://codeium.com) within the IDE, it would be great to be able to have this functionality.

Request support in Codeium Support channel: https://discord.com/channels/1027685395649015980/1088232046666727554

--- For Spanish speakers / Para hablantes de español ---

¡Hola! Soy usuario de NetBeans desde hace 15 años, me gustaría poder usar [Codeium AI](https://codeium.com) dentro del IDE, sería genial poder contar con esta funcionalidad.

**Comment by jimenezbre** [comment]
i would like it too

**Comment by Chris2011** [comment]
I see this as a plugin and not part of the IDE itself. Plugins are better in some reason to fast updates. Early the days we had also updates of packages within NetBenas but this seems dead under Apache due to some restrictions. Correct me when I'm wrong.

With the virtual-text that NetBeans has it could be possible to implement just a provider for any AI or much AI`s. I also thought of using the OpenAI Api and adding the localhost url to ollama to it, because the API is identical. So for this we could add codium also, but that needs time. One day when AI makes our jobs obsolete so that we can do what we ever wanted, I will have a look into this too.

**Comment by jean-luc-insset** [comment]
I would greatly appreciate a NetBeans plugin

**Comment by helderneres** [comment]
Codeium uses canny to manage feature requests. I have created a feature request to add support for Apache NetBeans. I think it would be ideal if the AI ​​tool would provide a plugin for the IDE. Please vote for this request.

https://codeium.canny.io/feature-requests/p/please-add-apache-netbeans-ide-support

**Comment by luckebar** [comment]
There is the plugin "Jeddict AI Assistant" that allows you to connect to any source that supports the OpenAI protocol. It should have similar functionality, I tried it and it does what it promises.
I personally do not like that an AI writes code or corrects my methods inside the IDE also because for simple things there is no need, for complex ones it will hardly give you back a code that perfectly adheres to your expectations at the first try.

**Reply by helderneres** [reply]
This plugin has been continuously updated. It provides support for multiple platforms/vendors. It looks very promising.

**Comment by manojkumarredbus** [comment]
Two more plugins are available for netBeans:

Original : https://github.com/Hillrunner2008/netbeans-chatgpt
Ollama based: https://github.com/manojkumarredbus/netbeans-Ollama
